sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -381,6 +382,64 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
     }
   }
 
+  private def prunePartitionSchema(partSchema: StructType, projectList: Seq[Expression],
+      filterCondition: Option[Expression] = None): StructType = {
+    // Luckily partition columns do not support nested types. So only need to prune the
+    // top level columns.
+    StructType(partSchema.filter(f =>
+      // The partition column used either in the projectList or the filterCondition will
+      // be kept.
+      (projectList ++ filterCondition).exists { expr =>
```

Comment:
This ends up concatenating the lists on each iteration of the schema elements.  Seems like we should be using AttributeSet here, if possible, constructed outside the filter for reuse on each element.

Reply:
Can you explain more about `Seems like we should be using AttributeSet ` ?

Reply:
Normally Spark uses `AttributeSet` to check if an attribute appears in another set of attributes rather than manually comparing names.  I think we're OK in this case, as worst case we'll assume a partition column is used when it isn't, and that's not query breaking, just inefficient, e.g.: a project after a scan where the scan does _not_ load any partition column but aliases another column to the partition column name, then followed by a filter on that aliased name.  Admittedly not very likely, but theoretically possible.  I'm OK with it as-is, as this case is pretty contrived and it should "fail safe" vs. not providing a partition column when it is needed.

Reply:
> e.g.: a project after a scan where the scan does not load any partition column but aliases another column to the partition column name, then followed by a filter on that aliased name. 

Thx for the details. Actually this is a reason I chose to use the raw project expression list instead of attributes here. This new aliased column will be a GpuAlias("b", ... AtrributeReference("other_col")), and it can be skipped becasue we only check the actual `AtrributeReference`s.